app/services/slack_notification_service.rb

```diff
@@ -46,6 +46,9 @@ def filter_opened_action
   end
 
   def find_message(text)
+    # If the ON HOLD label is quickly added and removed the search will not work because the message is not found by slack.
+    # We need to wait.
+    sleep(30)
```

Comment:
@MaicolBen Appreciate if you have an alternative for this solution.

Reply:
What I don´t like is that we are waiting for all request, the best option would be to just wait when the matches are empty. So an alternative is: you try the first time without sleep and if empty you sleep and call the query again
```
        if (matches.empty?)
       sleep(30)
       response = client_find.search_messages(channel: CHANNEL, query: "#{text} in:#{CHANNEL}")
       matches = response.dig(:messages, :matches)
     end
     matches ```

Reply:
you can delay the call with delayed job (just for simplicity, we don't care about performance here)

Reply:
Got it, that was my main concern. At least, we should take this last approach

Reply:
We were talking with @MaicolBen. delayed job needs some db or redis to work (not sure if we can add the heroku add on), another alternative is https://github.com/brandonhilkert/sucker_punch but both will requiere some time to configure and run. We can merge this with or without the sleep and deploy, so we can have the new features. Then with more time we can find a nice solution for delaying the jobs.